shared/private/header.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "system.h"
+
+namespace ink {
+		struct Header {
```

Comment:
Should be in snake_case (i.e. `header`)

https://github.com/brwarner/inkcpp/blob/master/CONTRIBUTING.md#code-style